source/data-formats/extended-json.txt

```diff
@@ -0,0 +1,103 @@
+.. _ruby-extended-json:
+
+=============
+Extended JSON
+=============
+
+.. facet::
+   :name: genre
+   :values: reference
+ 
+.. meta::
+   :keywords: code example, json, standard formatting
+   :description: Learn how to use extended JSON in the MongoDB Ruby Driver.
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 2
+   :class: twocols
+
+.. sharedinclude:: dbx/extended-json.rst
+
+Read Extended JSON
+------------------
+
+You can read an Extended JSON string into an Ruby array by calling
+the ``BSON::ExtJSON.parse()`` method. This method parses an Extended
+JSON string and returns an array containing the data.
+
+The following example shows how you can read an Extended JSON string into a
+array of hashes by using the ``parse()`` method:
+
+.. io-code-block::
+    :copyable:
+
+    .. input:: /includes/data-formats/extended-json.rb
+       :start-after: start-ex-json-read
+       :end-before: end-ex-json-read
+       :language: ruby
+       :dedent:
+
+   .. output::
+      :language: none
+      :visible: false
+
+      {"foo" => [1, 2]}
+      {"bar" => {"hello" => "world"}}
+      {"code" => #<BSON::CodeWithScope:0x0000000123f398e0 @javascript="function x() { return 1; }", @scope={}>}
+      {"bin" => <BSON::Binary:0x7144 type=user data=0x01020304...>}
+
+Write Extended JSON
+-------------------
+
+You can write an Extended JSON string from a hash by using the
+``as_extended_json()`` method. By default, this method returns
+the Extended JSON string in canonical format, but you can specify relaxed or
+legacy formats by passing a ``mode`` argument. 
+
+.. note:: Legacy Version
+
+   The legacy format option tells the {+driver-short+} to serialize BSON types
+   with the MongoDB Extended JSON v1 format, which predates the current
+   relaxed and canonical formats. 
+
+   For more information see, the :manual:`MongoDB Extended JSON v1
+   </reference/mongodb-extended-json-v1/>` page in the Server manual. 
+
+The following example outputs an Extended JSON string in the canonical, relaxed,
+and legacy formats. It calls the ``as_extended_json()`` method from inside a
+Ruby ``map{}`` block to create the Extended JSON string from an array of hashes:
```

Comment:
As noted in a later comment, you can invoke `#as_extended_json` directly on an array instance.

```suggestion
and legacy formats. It calls the ``#as_extended_json`` method on an array to create the
Extended JSON string from an array of hashes:
```

Also, Ruby convention refers to instance methods as `#method_name`, rather than `method_name()`.